pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -2102,6 +2110,13 @@ func reportGatewayStatus(
 		if len(addressesToReport) > 0 {
 			gs.Addresses = make([]k8sbeta.GatewayStatusAddress, 0, len(addressesToReport))
 			for _, addr := range addressesToReport {
+				// If the address is not an IP, then we check if it is a hostname
+				if _, err := netip.ParseAddr(addr); err != nil {
+					// TODO: this is a bit of a hack, we should probably have a better way to determine if an address is a hostname
+					if validHostnameRegexp.MatchString(addr) {
```

Comment:
Do we need this? Seems like if its not an IP its a hostname?

Reply:
If the regex doesn't match somehow then its not an IP and not a hostname. But we will try to write it as an IP address. So either way it gets rejected, just in one case we have a complex regex ? I think

Reply:
I initially tried to do it using only the regex but that didn't work for quite obvious reasons. It probably is a bit useless to double stack and have the regex here, but thought it might be more technically correct. I'm not really familiar with the code base so I figured exposing the error to the user might have a use for checking the hostname regex.